### PR TITLE
Support deletion prevention for `kubectl gs template app` command

### PR DESCRIPTION
**Pull request by AndiDog**
### What does this PR do?

https://github.com/giantswarm/giantswarm/issues/28310

Support [deletion prevention](https://docs.giantswarm.io/advanced/deletion-prevention/) through a new flag.

This is both helpful for customers and our own use (e.g. in MC bootstrapping where we typically don't want an MC to be deletable).

### What does it look like?

```
$ kubectl-gs template app --catalog=giantswarm --cluster-name="anditest" --name=nginx-ingress-controller-app --target-namespace=kube-system --version=2.30.0  --user-configmap <(echo "controller: { service: { public: false } }" | yq -y) --prevent-deletion
---
apiVersion: v1
data:
  values: |
    controller:
      service:
        public: false
kind: ConfigMap
metadata:
  creationTimestamp: null
  labels:
    giantswarm.io/prevent-deletion: "true"
  name: nginx-ingress-controller-app-userconfig-anditest
  namespace: anditest
---
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    giantswarm.io/prevent-deletion: "true"
  name: nginx-ingress-controller-app
  namespace: anditest
spec:
  catalog: giantswarm
  kubeConfig:
    inCluster: false
  name: nginx-ingress-controller-app
  namespace: kube-system
  userConfig:
    configMap:
      name: nginx-ingress-controller-app-userconfig-anditest
      namespace: anditest
  version: 2.30.0
```

### Do the docs need to be updated?

Yes, the article linked above.

### Should this change be mentioned in the release notes?

- [x] CHANGELOG.md has been updated

### Is this a breaking change?

No. The flag defaults to false.

### CODE REVIEW

File: cmd/template/app/runner.go [hunk 1]
```diff
@@ -147,6 +162,10 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
 	}
 	appConfig.NamespaceConfigLabels = namespaceLabels
 
+	if r.flag.PreventDeletion {
+		appConfig.ExtraLabels["giantswarm.io/prevent-deletion"] = "true"
```
Comment (whites11) [inline cmd/template/app/runner.go:166]: how about extracting the label to a constant?

File: cmd/login/runner_config_modify_test.go [hunk 2]
```diff
@@ -268,6 +268,7 @@ func TestKubeConfigModification(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
+		tc := tc
```
Comment (fiunchinho) [inline cmd/login/runner_config_modify_test.go:271]: How come is this now necessary?

**Comment by fiunchinho** [comment]
How are CAPI/CAPA/CAPZ/etc CRs protected? I was expecting we passed the value to the cluster-app so that it added the prevention label.

File: cmd/login/runner_config_modify_test.go (hunk 2 above)
Comment (AndiDog) [reply cmd/login/runner_config_modify_test.go:271]: Lint error of implicit memory aliasing. Newer Go versions finally get rid of this by making the loop variables locally-scoped, but for now we need this. My changes didn't lead to the build errors, but I wanted to fix them.

File: cmd/template/app/runner.go (hunk 1 above)
Comment (AndiDog) [reply cmd/template/app/runner.go:166]: Right. Let's add it in k8smetadata: https://github.com/giantswarm/k8smetadata/pull/83

**Comment by AndiDog** [comment]
> How are CAPI/CAPA/CAPZ/etc CRs protected? I was expecting we passed the value to the cluster-app so that it added the prevention label.

That needs to go into `kubectl gs template cluster`, and I think we should have that as well. Team Turtles wants to move shared stuff into a central cluster chart, so we may wait for that before going on with https://github.com/giantswarm/giantswarm/issues/28310. We can't just add the label to all resources in the `cluster-{aws,...}` chart since some need to be deletable on Helm upgrade (e.g. `name: mycluster-<hash>`).

I don't want `kubectl gs template app --prevent-deletion` to set some magic `.Values.metadata.preventDeletion=true` implicitly, but really only protect the `App`.

**Review by fiunchinho — APPROVED** [review]

**Review by mogottsch — APPROVED** [review]
Nice, thanks for making the prevent deletion mechanism more usable!

**Comment by AndiDog** [comment]
`go mod tidy` was missing. Auto-merging afterwards.

**Pull request merged**